main.cpp

```diff
@@ -54,48 +54,73 @@ Purpose:  This project will show you the difference between member functions and
 
 #include <iostream>
 #include <string>
+
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    T(int v, const char* n) : value(v), name(n) {}  
+    int value;                                     
+    std::string name;                              
 };
 
-struct <#structName1#>                                //4
+struct Comparison                                
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare( T* a, T* b ) 
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if( a && b )
+        {
+            if( a->value < b->value ) return a;
+            if( a->value > b->value ) return b;
+        }
+
         return nullptr;
     }
 };
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float x { 0 }, y { 0 };
+    
+    float modify( float* val_ ) 
     {
-        
+        if( val_ )
+        {
+            std::cout << "U's x value: " << x << std::endl;
+            x = *val_; 
+            std::cout << "U's x updated value: " << x << std::endl;
+            while( std::abs(y - x) > 0.001f )   
+                y += 0.1f ;
+            
+            std::cout << "U's y updated value: " << y << std::endl;
+            return y * x;
+        }
+        else
+        {
+            std::cout << "WARNING! nullptr detected. Task aborted!" << std::endl;
+            return 0;
+        }
     }
 };
 
-struct <#structname2#>
+struct Modulation
 {
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float modulate( U* that, float* val )
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        if( that != nullptr && val != nullptr )
         {
-            /*
-             write something that makes the distance between that-><#name2#> and that-><#name1#> get smaller
-             */
-            that-><#name2#> += ;
+            std::cout << "U's x value: " << that->x << std::endl;
+            that->x = *val;
+            std::cout << "U's x updated value: " << that->x << std::endl;
+            while( std::abs(that->y - that->x) > 0.001f )   
+                that->y += 0.1f ;
+
+            std::cout << "U's y updated value: " << that->y << std::endl; 
+            return that->y * that->x;
         }
-        std::cout << "U's <#name2#> updated value: " << that-><#name2#> << std::endl;
-        return that-><#name2#> * that-><#name1#>;
+        else
+        {
+            std::cout << "WARNING! nullptr detected. Task aborted!" << std::endl;
+            return 0;
+        }  
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
You do not need the `else` block.  Here's why:
`return` statements are the last thing that **ever happens** in a function.
When the `if()` expression is evaluated as `true` and the block is entered, the `return aValue;` is the last line of code to be executed before the function exits.
but when the `if()` expression evaluates as `false` and the block is **NOT** entered, the next thing that happens is `return anotherValue;`
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
that means you do **NOT** need the `else` block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    
    return anotherValue;
}
```
- [ ] revise